src/lib/destinations/webhook.ts

```diff
@@ -41,5 +61,52 @@ export const deliverToDestination = async (
     method: method ?? "POST",
     url: input.destination.url,
     status: response.status,
+    body: params.body,
   };
 };
+
+const buildWebhookInput = (
+  destinationPayload: DeliverToDestinationInput["destinationPayload"],
+  additionalInput?: object
+): string | object => {
+  if (additionalInput && typeof destinationPayload === "string") {
+    return {
+      payload: destinationPayload,
+      ...additionalInput,
+    };
+  } else if (additionalInput && typeof destinationPayload === "object") {
+    return {
+      ...destinationPayload,
+      ...additionalInput,
+    };
+  }
+  return destinationPayload;
+};
+
+const buildSource = async (
+  input: DeliverToDestinationInput & { destination: { type: "webhook" } }
+) => {
+  const transactionEventResult = TransactionEventSchema.safeParse(input.source);
```

Comment:
nit: IMO it would be a bit cleaner to keep this part of the logic in the `edi-inbound` handler -- that way the destination logic can remain dumb and just worry about the binary include/don't include source decision. 

I'm guessing you have it here so that we only go through the hassle of reconstructing the source transaction when the feature is enabled, which I also understand. too bad there's not a simpler way to get the "source transaction set" directly from Core